packages/renderers-js/src/fragments/discriminatorConstants.ts

```diff
@@ -81,27 +82,30 @@ export function getFieldDiscriminatorConstantFragment(
 
     const name = camelCase(`${prefix}_${discriminatorNode.name}`);
     const encoder = visit(field.type, typeManifestVisitor).encoder;
+    const strictType = visit(field.type, typeManifestVisitor).strictType;
     const value = visit(field.defaultValue, typeManifestVisitor).value;
-    return getConstantFragment({ ...scope, encoder, name, value });
+    return getConstantFragment({ ...scope, encoder, name, strictType, value });
 }
 
 function getConstantFragment(
     scope: Pick<GlobalFragmentScope, 'nameApi'> & {
         encoder: Fragment;
         name: string;
+        strictType: Fragment;
         value: Fragment;
     },
 ): Fragment {
-    const { encoder, name, nameApi, value } = scope;
+    const { encoder, name, nameApi, value, strictType } = scope;
     const constantName = nameApi.constant(name);
     const constantFunction = nameApi.constantFunction(name);
 
     return mergeFragments(
         [
-            mapFragmentContent(value, c => `export const ${constantName} = ${c};`),
+            mapFragmentContent(value, c => `export const ${constantName}: ${strictType.content} = ${c};`),
             mapFragmentContent(
                 encoder,
-                c => `export function ${constantFunction}() { return ${c}.encode(${constantName}); }`,
+                c =>
+                    `export function ${constantFunction}(): ReadonlyUint8Array { return ${c}.encode(${constantName}); }`,
```

Comment:
Whenever you introduce a new item that needs to be imported, it should be part of the fragment's import. Using the new Fragment API, you'll be able to do something like:

```ts
const readonlyUint8Array = use("ReadonlyUint8Array", "solanaCodecsCore");
return fragment`export function ${constantFunction}():${readonlyUint8Array} { return ${encoder}.encode(${constantName}); }`
```